### PR TITLE
Finally, here are some commits, and more to come...

### PR DESCRIPTION
**Pull request by terminalforlife**
See commit messages for details on the changes.

### CODE REVIEW

**Comment by huijunchen9260** [comment]
I am now testing...
FM_MVR is not malfunctioned. In the destination menu, we cannot properly choose the menu by selecting `./`. It ends up with an endless loop.

**Comment by terminalforlife** [comment]
Do you mean it _is_ malfunctioning since my commits? To which commit/file/line are you referring?

**Comment by terminalforlife** [comment]
Oh, it's probably because you've used some invalid syntax around 162, in the if statement. I don't know how to fix it, because I don't know what you were trying to do.

File: dmenufm [hunk 1]
```diff
@@ -21,159 +21,213 @@
...
+CheckDeps () {
+	DepCount=0
+	for CurDep in "$@"; do
+		if ! command -v "$CurDep" 1> /dev/null 2>&1; then
+			DepCount=$((DepCount + 1))
+			printf "ERROR: Dependency '$CurDep' not met." 1>&2
+		fi
+	done
+
+	[ $DepCount -eq 0 ] || exit 1
```
Comment (huijunchen9260) [inline dmenufm]: If you exit 1, then what's the point of printing error? In my case, the whole terminal ended, and user cannot see the error message.

Comment (terminalforlife) [reply dmenufm]: For debugging purposes.

**Comment by huijunchen9260** [comment]
I have adopted some commits and mix it into the newest commit. I really appreciate your contribution to dmenufm! Hope that you can give me some suggestion on my newest commit!

**Pull request closed without merging**